dao/src/main/scala/za/co/absa/enceladus/dao/rest/AuthClient.scala

```diff
@@ -62,7 +63,7 @@ sealed abstract class AuthClient(username: String, restTemplate: RestTemplate, a
           log.info(s"Authentication successful: $username")
           getAuthHeaders(response)
         case _             =>
-          throw UnauthorizedException(s"Authentication failure ($statusCode): $username")
+          throw AuthenticationException(s"Authentication failure ($statusCode): $username")
```

Comment:
fixed; I'll think about the refactoring again, there might be a more intelligent way how to do this

Reply:
I was trying to add more test into RestClientSuite.scala, but then I realised that this scenario is very difficult to simulate with unit tests without overwriting a lot of code (or writing a new unit test module(s)). 

I think that an integration test would be a much much better candidate for covering this scenario with tests, because it just uses/calls too many modules (RestClient, AuthClient, CrossHostApiCaller, and a few helper modules).